packages/vite/src/node/build.ts

```diff
@@ -613,6 +614,16 @@ function staticImportedByEntry(
   return someImporterIs
 }
 
+export function resolveLibFilename(
+  libOptions: LibraryOptions,
+  format: ModuleFormat | undefined,
+  pkgName: string
+): string {
+  return typeof libOptions.fileName === 'function' && format
+    ? libOptions.fileName(format)
+    : `${libOptions.fileName || pkgName}.${format || `es`}.js`
```

Comment:
And always expect a format here:
```suggestion
export function resolveLibFilename(
  libOptions: LibraryOptions,
  format: ModuleFormat,
  pkgName: string
): string {
  return typeof libOptions.fileName === 'function'
    ? libOptions.fileName(format)
    : `${libOptions.fileName || pkgName}.${format}.js`
```

Reply:
Makes sense to me! I added the changes accordingly